### DISCUSSION TITLE
Configurable base path?

### DISCUSSION THREAD

**Discussion by chromey** [featurehub-io/featurehub#631] [Q&A] [answered]
Hi there,

I might be missing something, but I found no way to configure feature hub to use a base path (context root) other than "/", which would be kind of a showstopper in our environment (and I guess many others as well?). Should I open a feature request for that or am I really just missing something?

**Comment by rvowles** [comment]
Hey there! Could you give us a little more context please? FeatureHub is designed to run and be hosted on its own URL, so being at context root is ok, so I suspect I am not understanding something? Generally we are happy to accomodate if possible!

**Comment by chromey** [comment]
Sure, thanks for getting back so quickly first of all.
We are sharing a platform with a bunch of other teams, and the teams and apps running on the platform are namespaced by context root, e.g I would like to deploy featurehub under `https://our-domain.com/our-team/featurehub`.
I assume your approach is to do something like `https://featurehub.our-domain.com/`? ... but we cannot easily create new sub-domains.
I naively attempted to do some URL re-writing in our service mesh, but that doesn't work because, for example, the admin web UI uses absolute paths.

**Comment by IrinaSouth** [comment]
Hi @chromey, just to let you know, FeatureHub is designed to support an enterprise with multiple teams. We have a concept in our admin app, where a super admin can create Portfolios (this could be your teams).  Once you setup Portfolios you can add multiple applications inside of each Portfolio. There is a concept of Portfolio Admin and also user groups that can have access only to specified portfolios and even more granular per application access. Have you thought of splitting your FeatureHub setup like this? I think in this case you may not need to do the namespacing. But if this is still not suitable for your use case, let us know. Cheers!

**Comment by chromey** [comment]
Hi @IrinaSouth, yes, I read about the portfolio feature and I think it will become handy if and when other teams want to adopt FeatureHub as well (right now i'ts just us). But even then we would deploy it on the same domain as our other shared tools. For example we have `our-domain.com/tools/jenkins`, `our-domain.com/tools/nexus` and so on, and we would not want to break away from that pattern. I don't think this is an uncommon approach, and the fact that all our other self-hosted tools (that I know of) support configuring a context root seems to prove my point.

Let me also say that I'm very impressed with FeatureHub's capabilities, it would be a real shame if this were the reason why we could not use it in the end.

**Comment by IrinaSouth** [comment]
Hey @chromey , thank you for explaining your use case, I have opened an issue based on your request https://github.com/featurehub-io/featurehub/issues/634 and we will look at resolving it soon. Glad you find FeatureHub useful, and let us know if anything else stands out, we are always after our customer's feedback.

**Comment by chromey** [comment]
Excellent, will keep an eye on it! Thanks a lot, and keep up the great work!

**Comment by IrinaSouth** [comment, marked as answer]
This is now released in FeatureHub 1.5.5-RC1.
